File: з2.pas
program one;
const 
  N = 8;
type 
  m1 = array[1..N, 1..N] of integer;
  m2 = array[1..N] of integer;

procedure random(var matrix: m1);
var 
  i, j: integer;
begin
  for i := 1 to N do
    for j := 1 to N do
      matrix[i, j] := random(7);
end;

procedure print(a: m1);
var 
  i, j: integer;
begin
  writeln();
  for i := 1 to N do
  begin
    for j := 1 to N do
      write(a[i, j]:5);
    writeln;
  end;
end;

function para7(row: m2): boolean;
var 
  j: integer;
begin
  para7 := false;
  for j := 1 to N - 1 do
    if row[j] + row[j + 1] = 7 then
    begin
      para7 := true;
      exit;
    end;
end;

procedure res(b: m1; var r: m2);
var 
  i, j: integer; // добавил j здесь
  temp: m2;
begin
  for i := 1 to N do
  begin
    for j := 1 to N do
      temp[j] := b[i, j];
    
    if para7(temp) then
      r[i] := 1
    else
      r[i] := -1;
  end;
end;

procedure resultat(res: m2);
var 
  i: integer;
begin
  writeln('результат');
  for i := 1 to N do
    write(res[i]:5);
  writeln;
end;
var 
  matrix: m1;
  matrix2: m2;
begin
  randomize;
  random(matrix);
  print(matrix);
  res(matrix, matrix2);
  resultat(matrix2);
end.

File: з2.py
import random
N = 8
def random_matrix(matrix):
    for i in range(N):
        for j in range(N):
            matrix[i][j] = random.randint(0, 6)

def print_matrix(a):
    print()
    for i in range(N):
        for j in range(N):
            print(f'{a[i][j]:5}', end='')
        print()

def para7(row):
    for j in range(N - 1):
        if row[j] + row[j + 1] == 7:
            return True
    return False

def res(b, r):
    temp = [0] * N
    for i in range(N):
        for j in range(N):
            temp[j] = b[i][j]

        if para7(temp):
            r[i] = 1
        else:
            r[i] = -1

def resultat(res_arr):
    print('результат')
    for i in range(N):
        print(f'{res_arr[i]:5}', end='')
    print()

matrix = [[0] * N for _ in range(N)]
random_matrix(matrix)
print_matrix(matrix)
matrix2 = [0] * N
res(matrix, matrix2)
resultat(matrix2)